index.js

```diff
@@ -28,12 +28,18 @@
      * and a size, which represents the number of nodes in the list.
      *
      * @constructor
+     * @param {array} arr The data to initialize with the list
      */
-    function LinkedList() {
+    function LinkedList(arr) {
         this.head = null;
         this.tail = null;
         this.size = 0;
-
+        // insert initial nodes
+        if(arr && arr instanceof Array && arr.length > 0){            
+            for(var i = 0; i < arr.length; i++){
+                this.insert(arr[i]);
+            }
+        }
```

Comment:
Typically, I have a preference for utilizing a `forEach` to loop over arrays; however, this will certainly work.  And, in this case, we would need to ensure we have the right `this` context if we did make use of a `forEach` so this makes sense here.